pkgs/build-support/rust/cargo-vendor-normalise.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+
+import sys
+
+import toml
+
+
+def quote(s: str) -> str:
+    escaped = s.replace('"', r"\"").replace("\n", r"\n").replace("\\", "\\\\")
+    return '"{}"'.format(escaped)
+
+
+def main() -> None:
```

Comment:
I added type checking here and reformatted the code with `black` + some minor readability improvements.